src/bunit.web.anglesharp/WrapperElementsGenerator.cs

```diff
@@ -0,0 +1,136 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Bunit.Web.AngleSharp;
+
+[Generator]
+public class WrapperElementsGenerator : IIncrementalGenerator
+{
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		// Finds the AngleSharp assembly referenced by the target project
+		// This should prevent the source generator from running unless a
+		// new symbol is returned.
+		var angleSharpAssemblyReference = context
+			.CompilationProvider
+			.Select((compilation, cancellationToken) =>
+			{
+				var meta = compilation.References.FirstOrDefault(x => x.Display?.EndsWith($"{Path.DirectorySeparatorChar}AngleSharp.dll", StringComparison.Ordinal) ?? false);
+				return compilation.GetAssemblyOrModuleSymbol(meta);
+			});
+
+		// Output the hardcoded source files
+		context.RegisterSourceOutput(angleSharpAssemblyReference, GenerateStaticContent);
+
+		// Output the generated wrapper types
+		context.RegisterSourceOutput(angleSharpAssemblyReference, GenerateWrapperTypes);
+	}
+
+	private static void GenerateStaticContent(SourceProductionContext context, ISymbol assembly)
+	{
+		if (assembly is not IAssemblySymbol)
+			return;
+
+		var elementFactorySource = ReadEmbeddedResource("Bunit.Web.AngleSharp.IElementFactory.cs");
+		var wrapperBase = ReadEmbeddedResource("Bunit.Web.AngleSharp.WrapperBase.cs");
+		context.AddSource("IElementFactory.g.cs", elementFactorySource);
+		context.AddSource("WrapperBase.g.cs", wrapperBase);
+	}
+
+	private static void GenerateWrapperTypes(SourceProductionContext context, ISymbol assembly)
+	{
+		if (assembly is not IAssemblySymbol angleSharpAssembly)
+			return;
+
+		var elementInterfacetypes = FindElementInterfaces(angleSharpAssembly);
+
+		var source = new StringBuilder();
```

Comment:
Small perf optimization: Initialize with a pre-allocated buffer.
```suggestion
        var source = new StringBuilder(2000);
```